x-pack/legacy/plugins/fleet/public/components/policy_form.tsx

```diff
@@ -4,42 +4,74 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
+import React, { useState } from 'react';
 import { EuiFieldText, EuiForm, EuiFormRow } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
 import { Policy } from '../../common/types/domain_data';
 
+interface ValidationResults {
+  [key: string]: React.ReactNode[];
+}
+
+export const policyFormValidation = (policy: Partial<Policy>): ValidationResults => {
+  const errors: ValidationResults = {};
+
+  if (!policy.name?.trim()) {
+    errors.name = [
+      <FormattedMessage
+        id="xpack.fleet.policyForm.nameRequiredErrorMessage"
+        defaultMessage="Policy name is required"
+      />,
+    ];
+  }
+
+  return errors;
+};
+
 interface Props {
   policy: Partial<Policy>;
   updatePolicy: (u: Partial<Policy>) => void;
+  validation: ValidationResults;
 }
 
-export const PolicyForm: React.FC<Props> = ({ policy, updatePolicy }) => {
+export const PolicyForm: React.FC<Props> = ({ policy, updatePolicy, validation }) => {
+  const [touchedFields, setTouchedFields] = useState<{ [key: string]: boolean }>({});
+
   return (
     <EuiForm>
-      {/* Name */}
-      <EuiFormRow
-        label={
-          <FormattedMessage id="xpack.fleet.policyForm.nameFieldLabel" defaultMessage="Name" />
-        }
-      >
-        <EuiFieldText value={policy.name} onChange={e => updatePolicy({ name: e.target.value })} />
-      </EuiFormRow>
-
-      {/* Description */}
-      <EuiFormRow
-        label={
+      {[
```

Comment:
I think this would be a bit cleaner and easier to read if stored in a descriptive variable and then mapped through.